server/src/main/java/org/elasticsearch/common/lucene/search/function/ScriptScoreFunction.java

```diff
@@ -62,32 +62,41 @@ public LeafScoreFunction getLeafScoreFunction(LeafReaderContext ctx) throws IOEx
         leafScript._setIndexName(indexName);
         leafScript._setShard(shardId);
         return new LeafScoreFunction() {
-            @Override
-            public double score(int docId, float subQueryScore) throws IOException {
+
+            private double score(int docId, float subQueryScore, ScoreScript.ExplanationHolder holder) throws IOException {
                 leafScript.setDocument(docId);
                 scorer.docid = docId;
                 scorer.score = subQueryScore;
-                double result = leafScript.execute(null);
+                double result = leafScript.execute(holder);
+
                 if (result < 0f) {
                     throw new IllegalArgumentException("script score function must not produce negative scores, but got: [" + result + "]");
                 }
                 return result;
             }
 
+            @Override
+            public double score(int docId, float subQueryScore) throws IOException {
+                return score(docId, subQueryScore, null);
+            }
+
             @Override
             public Explanation explainScore(int docId, Explanation subQueryScore) throws IOException {
                 Explanation exp;
                 if (leafScript instanceof ExplainableScoreScript) {
-                    leafScript.setDocument(docId);
-                    scorer.docid = docId;
-                    scorer.score = subQueryScore.getValue().floatValue();
                     exp = ((ExplainableScoreScript) leafScript).explain(subQueryScore);
```

Comment:
I am not sure why this was removed, but it should be added back.

Reply:
Ah yes, you just beat me to it; that's why the build was failing.  Misunderstanding on my part.  It's added back now.